react-app/ccf/src/functions/storage.ts

```diff
@@ -0,0 +1,58 @@
+import { ref, uploadBytes, getDownloadURL, listAll } from "firebase/storage";
+import { storage } from '../index'
+
+export const uploadFileToStorage = async (file: File): Promise<string> => {
+    if (!file) {
+      throw new Error('No file selected');
+    }
+  
+    const storageRef = ref(storage, 'pdfs/' + file.name);
+  
+    try {
+      const snapshot = await uploadBytes(storageRef, file);
+      console.log('Uploaded a blob or file!');
+      
+      const downloadURL = await getDownloadURL(snapshot.ref);
+      console.log('File available at', downloadURL);
+      
+      return 'File uploaded successfully!';
+    } catch (error) {
+      console.error('Upload failed', error);
+      throw error;
+    }
+  };
+
+export const downloadSpecificPDFS = async (arr: string[]): Promise<Array<{ name: string, url: string }>> => {
```

Comment:
This function looks like it's doing extra work that it doesn't need to. Let me know if I'm missing something, but the PDFs should be retrievable just by their file name without getting a list of every PDF. https://firebase.google.com/docs/storage/web/download-files. Also, let's try to be more clear with function and variable names: "downloadPDFsByName" and "names" would be much more clear than "downloadSpecificPDFs" and "arr"